fails to install required packages on badfish venv

When trying to install the requirements for Badfish using Jetpack on RHEL 7.8 i.e the ansible jump host https://github.com/redhat-performance/jetpack/blob/master/tasks/install_packages.yml#L37, I run into this pip issue 
```
TASK [set badfish venv path] ******************************************************************************************
ok: [localhost]

TASK [Create badfish virtualenv] **************************************************************************************
ok: [localhost]

TASK [Install badfish requirements] ***********************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "cmd": ["/root/badfish/.venv/bin/pip", "install", "-r", "/root/badfish/requirements.txt"], "msg": "stdout: Requirement already satisfied: pyyaml>=4.2b1 in /root/badfish/.venv/lib/python3.6/site-packages (from -r /root/badfish/requirements.txt (line 6))\nCollecting aiohttp==3.6.2 (from -r /root/badfish/requirements.txt (line 7))\n  Using cached https://files.pythonhosted.org/packages/7c/39/7eb5f98d24904e0f6d3edb505d4aa60e3ef83c0a58d6fe18244a51757247/aiohttp-3.6.2-cp36-cp36m-manylinux1_x86_64.whl\nCollecting async-lru==1.0.2 (from -r /root/badfish/requirements.txt (line 8))\n  Using cached https://files.pythonhosted.org/packages/7e/c1/a3d6207deaaeb582d16dc9a0fd217f192efc9487ce59897131cf9a2bdc1c/async_lru-1.0.2.tar.gz\nCollecting pytest~=4.3.0 (from -r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/c8/52/c455e718f19e4fd7126c81b4910ca2431ded6c28b97e36a1a1f5e5ef7247/pytest-4.3.1-py2.py3-none-any.whl\nCollecting asynctest~=0.13.0 (from -r /root/badfish/requirements.txt (line 10))\n  Using cached https://files.pythonhosted.org/packages/e8/b6/8d17e169d577ca7678b11cd0d3ceebb0a6089a7f4a2de4b945fe4b1c86db/asynctest-0.13.0-py3-none-any.whl\nCollecting setuptools~=46.1.3 (from -r /root/badfish/requirements.txt (line 11))\n  Using cached https://files.pythonhosted.org/packages/a0/df/635cdb901ee4a8a42ec68e480c49f85f4c59e8816effbf57d9e6ee8b3588/setuptools-46.1.3-py3-none-any.whl\nRequirement already satisfied: multidict<5.0,>=4.5 in /root/badfish/.venv/lib/python3.6/site-packages (from aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\nCollecting yarl<2.0,>=1.0 (from aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\n  Using cached https://files.pythonhosted.org/packages/6d/0b/44006eb5b3b7898e54df4b7481839d22d15cf3fc4f0420394ad9e3bb214e/yarl-1.6.2.tar.gz\nRequirement already satisfied: async-timeout<4.0,>=3.0 in /root/badfish/.venv/lib/python3.6/site-packages (from aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\nRequirement already satisfied: attrs>=17.3.0 in /root/badfish/.venv/lib/python3.6/site-packages (from aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\nRequirement already satisfied: idna-ssl>=1.0; python_version < \"3.7\" in /root/badfish/.venv/lib/python3.6/site-packages (from aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\nRequirement already satisfied: chardet<4.0,>=2.0 in /root/badfish/.venv/lib/python3.6/site-packages (from aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\nRequirement already satisfied: typing-extensions>=3.6.5; python_version < \"3.7\" in /root/badfish/.venv/lib/python3.6/site-packages (from aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\nCollecting more-itertools>=4.0.0; python_version > \"2.7\" (from pytest~=4.3.0->-r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/ec/f7/9c33f38577d9fb9b1aa60c0fd9961d09574f5d47d2b830e5183e6adfc8cc/more_itertools-8.5.0-py3-none-any.whl\nCollecting py>=1.5.0 (from pytest~=4.3.0->-r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/68/0f/41a43535b52a81e4f29e420a151032d26f08b62206840c48d14b70e53376/py-1.9.0-py2.py3-none-any.whl\nCollecting atomicwrites>=1.0 (from pytest~=4.3.0->-r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/2c/a0/da5f49008ec6e9a658dbf5d7310a4debd397bce0b4db03cf8a410066bb87/atomicwrites-1.4.0-py2.py3-none-any.whl\nCollecting pluggy>=0.7 (from pytest~=4.3.0->-r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/a0/28/85c7aa31b80d150b772fbe4a229487bc6644da9ccb7e427dd8cc60cb8a62/pluggy-0.13.1-py2.py3-none-any.whl\nCollecting six>=1.10.0 (from pytest~=4.3.0->-r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/ee/ff/48bde5c0f013094d729fe4b0316ba2a24774b3ff1c52d924a8a4cb04078a/six-1.15.0-py2.py3-none-any.whl\nRequirement already satisfied: idna>=2.0 in /root/badfish/.venv/lib/python3.6/site-packages (from yarl<2.0,>=1.0->aiohttp==3.6.2->-r /root/badfish/requirements.txt (line 7))\nCollecting importlib-metadata>=0.12; python_version < \"3.8\" (from pluggy>=0.7->pytest~=4.3.0->-r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/6d/6d/f4bb28424bc677bce1210bc19f69a43efe823e294325606ead595211f93e/importlib_metadata-2.0.0-py2.py3-none-any.whl\nCollecting zipp>=0.5 (from importlib-metadata>=0.12; python_version < \"3.8\"->pluggy>=0.7->pytest~=4.3.0->-r /root/badfish/requirements.txt (line 9))\n  Using cached https://files.pythonhosted.org/packages/41/ad/6a4f1a124b325618a7fb758b885b68ff7b058eec47d9220a12ab38d90b1f/zipp-3.4.0-py3-none-any.whl\nInstalling collected packages: yarl, aiohttp, async-lru, more-itertools, py, atomicwrites, zipp, importlib-metadata, pluggy, six, setuptools, pytest, asynctest\n  Running setup.py install for yarl: started\n    Running setup.py install for yarl: finished with status 'error'\n    Complete output from command /root/badfish/.venv/bin/python3 -u -c \"import setuptools, tokenize;__file__='/tmp/pip-build-huwfnsjb/yarl/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\\r\\n', '\\n');f.close();exec(compile(code, __file__, 'exec'))\" install --record /tmp/pip-_t6in_8m-record/install-record.txt --single-version-externally-managed --compile --install-headers /root/badfish/.venv/include/site/python3.6/yarl:\n    **********************\n    * Accellerated build *\n    **********************\n    running install\n    running build\n    running build_py\n    creating build\n    creating build/lib.linux-x86_64-3.6\n    creating build/lib.linux-x86_64-3.6/yarl\n    copying yarl/__init__.py -> build/lib.linux-x86_64-3.6/yarl\n    copying yarl/_quoting.py -> build/lib.linux-x86_64-3.6/yarl\n    copying yarl/_quoting_py.py -> build/lib.linux-x86_64-3.6/yarl\n    copying yarl/_url.py -> build/lib.linux-x86_64-3.6/yarl\n    running egg_info\n    writing yarl.egg-info/PKG-INFO\n    writing dependency_links to yarl.egg-info/dependency_links.txt\n    writing requirements to yarl.egg-info/requires.txt\n    writing top-level names to yarl.egg-info/top_level.txt\n    reading manifest file 'yarl.egg-info/SOURCES.txt'\n    reading manifest template 'MANIFEST.in'\n    warning: no previously-included files matching '*.pyc' found anywhere in distribution\n    warning: no previously-included files matching '*.cache' found anywhere in distribution\n    warning: no previously-included files found matching 'yarl/*.html'\n    warning: no previously-included files found matching 'yarl/*.so'\n    warning: no previously-included files found matching 'yarl/*.pyd'\n    no previously-included directories found matching 'docs/_build'\n    writing manifest file 'yarl.egg-info/SOURCES.txt'\n    copying yarl/__init__.pyi -> build/lib.linux-x86_64-3.6/yarl\n    copying yarl/_quoting_c.c -> build/lib.linux-x86_64-3.6/yarl\n    copying yarl/_quoting_c.pyi -> build/lib.linux-x86_64-3.6/yarl\n    copying yarl/_quoting_c.pyx -> build/lib.linux-x86_64-3.6/yarl\n    copying yarl/py.typed -> build/lib.linux-x86_64-3.6/yarl\n    running build_ext\n    building 'yarl._quoting_c' extension\n    creating build/temp.linux-x86_64-3.6\n    creating build/temp.linux-x86_64-3.6/yarl\n    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/root/badfish/.venv/include -I/usr/include/python3.6m -c yarl/_quoting_c.c -o build/temp.linux-x86_64-3.6/yarl/_quoting_c.o\n    yarl/_quoting_c.c:4:20: fatal error: Python.h: No such file or directory\n     #include \"Python.h\"\n                        ^\n    compilation terminated.\n    error: command 'gcc' failed with exit status 1\n    \n    ----------------------------------------\n\n:stderr: Command \"/root/badfish/.venv/bin/python3 -u -c \"import setuptools, tokenize;__file__='/tmp/pip-build-huwfnsjb/yarl/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\\r\\n', '\\n');f.close();exec(compile(code, __file__, 'exec'))\" install --record /tmp/pip-_t6in_8m-record/install-record.txt --single-version-externally-managed --compile --install-headers /root/badfish/.venv/include/site/python3.6/yarl\" failed with error code 1 in /tmp/pip-build-huwfnsjb/yarl/\nYou are using pip version 9.0.3, however version 20.2.4 is available.\nYou should consider upgrading via the 'pip install --upgrade pip' command.\n"}

PLAY RECAP ************************************************************************************************************
localhost                  : ok=6    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```
Note: It has both python2 and python3 installed

Solution:
Upgrade the pip inside the badfish venv ```pip install --upgrade pip```

We hit this issue as it is trying to use the old pip version which is not suitable for Python 3.x venv’s

--- a/tasks/install_packages.yml
+++ b/tasks/install_packages.yml
@@ -34,6 +34,12 @@
   - name: Create badfish virtualenv
     command: python3 -m venv {{ badfish_venv }} creates={{ badfish_venv }}
 
+  - name: Upgrade pip
+    pip:
+      name: pip
+      extra_args: --upgrade
+      virtualenv: "{{ badfish_venv }}"
+
   - name: Install badfish requirements
     pip:
       requirements: "{{ ansible_user_dir }}/badfish/requirements.txt"

